src/UCommerce.Transactions.Payments.Quickpay/Pipelines/Initialize/QuickpayExtendedPaymentDefinitionTask.cs

```diff
@@ -0,0 +1,131 @@
+using Ucommerce.Pipelines;
+using Ucommerce.Pipelines.Initialization;
+using Ucommerce.Security;
+
+namespace Ucommerce.Transactions.Payments.Quickpay.Pipelines.Initialize
+{
+    public class QuickpayExtendedPaymentDefinitionTask : IPipelineTask<InitializeArgs>
+    {
+        private readonly IRepository<Definition> _definitionRepository;
+        private readonly IRepository<DefinitionType> _definitionTypeRepository;
+        private readonly IRepository<DataType> _dataTypeRepository;
+        private readonly ILoggingService _logger;
+        private readonly string _definitionName = "Quickpay";
+        private readonly int _definitionTypeId = 4;
+        private readonly int _shortTextDataType = 1;
+        private readonly int _booleanDataType = 6;
+
+        public QuickpayExtendedPaymentDefinitionTask(IRepository<Definition> definitionRepository, IRepository<DefinitionType> definitionTypeRepository, IRepository<DataType> dataTypeRepository, ICurrentUserNameService currentUserNameService, ILoggingService logger)
+        {
+            _definitionRepository = definitionRepository;
+            _definitionTypeRepository = definitionTypeRepository;
+            _dataTypeRepository = dataTypeRepository;
+            _logger = logger;
+        }
+
+        public PipelineExecutionResult Execute(InitializeArgs subject)
+        {
+            Definition definition = _definitionRepository.Select().FirstOrDefault(d => d.Name.Equals(_definitionName));
+            if (definition != null)
+            {
+                definition.Deleted = false;
+            }
+            else
+            {
+                definition = new Definition()
+                {
+                    Name = _definitionName,
+                    Description = "Configuration for Quickpay",
+                    DefinitionType = _definitionTypeRepository.Get(_definitionTypeId)
+                };
+                this._logger.Information<QuickpayExtendedPaymentDefinitionTask>("Quickpay payment definition created.");
+            }
+
+            var shortTextDataType = _dataTypeRepository.Get(_shortTextDataType);
+            var booleanDataType = _dataTypeRepository.Get(_booleanDataType);
+
+            this.CreateOrUpdateDefinitionField(definition, "ApiKey", shortTextDataType);
+            this.CreateOrUpdateDefinitionField(definition, "PrivateAccountKey", shortTextDataType);
+            this.CreateOrUpdateDefinitionField(definition, "Merchant", shortTextDataType, "12345678");
+            this.CreateOrUpdateDefinitionField(definition, "AgreementId", shortTextDataType, "12345678");
+            this.CreateOrUpdateDefinitionField(definition, "CallbackUrl", shortTextDataType, "(auto)");
+            this.CreateOrUpdateDefinitionField(definition, "Language", shortTextDataType, "(auto)");
+            this.CreateOrUpdateDefinitionField(definition, "AcceptUrl", shortTextDataType);
+            this.CreateOrUpdateDefinitionField(definition, "CancelUrl", shortTextDataType);
```

Comment:
@MadsMagnusDue like we talked about regarding Adyen, maybe the `AcceptUrl` and `CancelUrl` should vary by culture? Theretically also `Language` if value doesn't match culture/iso code.